src/com/intellij/idea/plugin/hybris/runConfigurations/LocalSapCXRunProfileState.kt

```diff
@@ -0,0 +1,125 @@
+    environment: ExecutionEnvironment, val project: Project
+) : CommandLineState(environment), RemoteState {
+
+    private val SHMEM_ADDRESS: String = "javadebug"
+    private val HOST = "localhost"
+    private val PORT = "8000"
+    private val SERVER_MODE = false
+    private val USE_SOCKET_TRANSPORT = true
+
+    init {
+        updateDebugPort(PORT)
+    }
+
+    private fun getScriptPath(): String {
+        val basePath = project.basePath!!
+        val settings = HybrisProjectSettingsComponent.getInstance(project).state
+        val hybrisDirectory = settings.hybrisDirectory!!
+        val script = if (SystemUtils.IS_OS_WINDOWS) HybrisConstants.HYBRIS_SERVER_BASH_SCRIPT_NAME else HybrisConstants.HYBRIS_SERVER_SHELL_SCRIPT_NAME
+
+        return Paths.get(basePath, hybrisDirectory, script).toString()
+    }
+
+    private fun getWorkDirectory(): String {
+        val basePath = project.basePath!!
+        val settings = HybrisProjectSettingsComponent.getInstance(project).state
+        val hybrisDirectory = settings.hybrisDirectory!!
+
+        return Paths.get(basePath, hybrisDirectory, HybrisConstants.PLATFORM_MODULE_PREFIX).toString()
+    }
+
+    override fun startProcess(): ProcessHandler {
+        val commandLine = GeneralCommandLine(getScriptPath())
+        commandLine.setWorkDirectory(getWorkDirectory())
+        if (executor is DefaultDebugExecutor) {
+            commandLine.addParameter("debug")
+        }
+        val processHandler = ProcessHandlerFactory.getInstance().createColoredProcessHandler(commandLine)
+        ProcessTerminatedListener.attach(processHandler)
+        return processHandler
+    }
+
+    private fun getRemoteConnection(propertyString: String): RemoteConnection {
+        var useSocket: Boolean = USE_SOCKET_TRANSPORT
+        var hostName: String = HOST
+        var address: String = PORT
+        var serverMode: Boolean = SERVER_MODE
+
+        for (setting in propertyString.split(',')) {
+            val part = setting.split('=')
+            when (part[0]) {
+                "transport" -> useSocket = part[1] == "dt_socket"
+                "server" -> serverMode = part[1] == "y"
+                "address" -> address = part[1]
+                "hostName" -> hostName = part[1]
+            }
+        }
+
+        val debugPort = if (useSocket) address else SHMEM_ADDRESS
+        updateDebugPort(debugPort)
+
+        return RemoteConnection(useSocket, hostName, debugPort, !serverMode)
+    }
+
+    private fun updateDebugPort(debugPort: String) {
+        val debuggerRunnerSettings = environment.runnerSettings as GenericDebuggerRunnerSettings
+        debuggerRunnerSettings.debugPort = debugPort
+    }
+
+    override fun getRemoteConnection(): RemoteConnection {
+        val propertyService = PropertyService.getInstance(project)
+        if (propertyService != null) {
+            val debugOptions = propertyService.findProperty(HybrisConstants.TOMCAT_JAVA_DEBUG_OPTIONS)!!
```

Comment:
mby fallback to default remote connection settings in such a case?
`!!` is risky